Patterns/Italian/Italian-DateTime.yaml

```diff
@@ -1,231 +1,286 @@
 FollowedDateUnit: !nestedRegex
   def: ^\s*{DateUnitRegex}
   references: [ DateUnitRegex ]
 NumberCombinedWithDateUnit: !nestedRegex
   def: \b(?<num>\d+(\.\d*)?){DateUnitRegex}
   references: [ DateUnitRegex ]
 QuarterRegex: !nestedRegex
-  def: (the\s+)?(?<cardinal>primo|secondo|terzo|quarto)\s+trimestre(\s+of|\s*,\s*)?\s+({YearRegex}|{RelativeRegex}\s+year)
-  references: [ YearRegex, RelativeRegex ]
+  def: ((il|l')\s*)?(?<cardinal>primo|secondo|terzo|quarto|ultimo)\s+trimestre((\s+(di|del(l[o'])?)|\s*,\s*)?\s*({YearRegex}|({RelativeRegex}\s*anno)|(anno\s+(({NextSuffixRegex})|({PastSuffixRegex})))))?
+  references: [ YearRegex, RelativeRegex, NextSuffixRegex, PastSuffixRegex ]
 QuarterRegexYearFront: !nestedRegex
-  def: ({YearRegex}|l'anno\s+({PastSuffixRegex}|{NextSuffixRegex})|{RelativeRegex}\s+anno)\s+(le\s+)?(?<cardinal>primo|secondo|terzo|quarto)\s+trimestre
+  def: ({YearRegex}|l'anno\s+({PastSuffixRegex}|{NextSuffixRegex})|{RelativeRegex}\s+anno)\s+(il\s+)?(?<cardinal>primo|secondo|terzo|quarto)\s+trimestre
   references: [ YearRegex, RelativeRegex, PastSuffixRegex, NextSuffixRegex ]
-AllHalfYearRegex: !simpleRegex
-  # TODO: modify below regex according to the counterpart in English
-  def: ^[.]
+AllHalfYearRegex: !nestedRegex
+  def: (la\s+)?((prima|seconda)\s+metà)(\s+(di|del(l[o'])?)|\s*,\s*)\s*({RelativeRegex}?\s*anno)
+  references: [ RelativeRegex ]
 PrefixDayRegex: !simpleRegex
-  # TODO: modify below regex according to the counterpart in English
-  def: ^[.]
+  def: \b(((inizio|metà|fine)((\s+(della\s+)?giornata)|(\s+del\s+giorno)))|((prima|(più\s+tardi))\s+quel\s+giorno))$
 CenturySuffixRegex: !simpleRegex
-  # TODO: modify below regex according to the counterpart in English
-  def: ^[.]
+  def: (^secolo)\b
 SeasonRegex: !nestedRegex
-  def: \b((<seas>primavera|estate|autunno|inverno)+\s*({NextSuffixRegex}|{PastSuffixRegex}))|(?<season>({RelativeRegex}\s+)?(?<seas>primavera|estate|autunno|inverno)((\s+del|\s*,\s*)?\s+({YearRegex}|{RelativeRegex}\s+l'anno))?)\b
+  def: \b((l[a']\s*)?(<seas>primavera|estate|autunno|inverno)\s*({NextSuffixRegex}|{PastSuffixRegex}))|(?<season>({RelativeRegex}\s*)?(?<seas>primavera|estate|autunno|inverno)((\s+(di|del(l[o'])?)|\s*,\s*)?\s*({YearRegex}|{RelativeRegex}\s*anno))?)\b
   references: [ YearRegex, RelativeRegex, NextSuffixRegex, PastSuffixRegex ]
 WhichWeekRegex: !simpleRegex
   def: \b(settimana)(\s*)(?<number>5[0-3]|[1-4]\d|0?[1-9])\b
 WeekOfRegex: !simpleRegex
-  def: (settimana)(\s*)(del)
+  def: (settimana)(\s*)(d[i']|del(l[aeo'])?|degli)
 MonthOfRegex: !simpleRegex
-  def: (mese)(\s*)(di)
-MonthRegex: !simpleRegex
-  def: (?<month>Aprile|Apr|Agosto|Ago|Dicembre|Dic|Febbraio|Feb|Gennaio|Gen|Luglio|Lug|Giugno|Giu|Marzo|Mar|Maggio|Mar|Novembre|Nov|Ottobre|Ott|Settembre|Set)
+  def: (mese)(\s*)(d[i']|del(l[aeo'])?|degli)
 #This is a look-behind assertion. Some cases should extract two digits as year like 11/25/16, where 16 means 2016. 
```

Comment:
Move these comments with the corresponding code.

Reply:
updated.